torch/_inductor/codecache.py

```diff
@@ -1717,6 +1717,11 @@ def get_include_and_linking_paths(
         else:
             libs = ["omp"] if config.is_fbcode() else ["gomp"]
 
+        # For AOT mode, the produced library relies on torch cpu to set grad mode
+        # like aoti_torch_grad_mode_set_enabled
+        if aot_mode and sys.platform == "linux" and not config.is_fbcode():
```

Comment:
@desertfire , this PR encountered an issue that could not find libtorch on MacOS Arm64. So I added more conditions here. Could you help review if the conditions make sense?